wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift

```diff
@@ -337,20 +278,79 @@ extension ConversationCreationController: AddParticipantsConversationCreationDel
             values.participants = users
 
         case .create:
-            var allParticipants = values.participants
-            allParticipants.insert(selfUser)
+            guard let userSession = ZMUserSession.shared() else { return }
+            let service = ConversationService(context: userSession.viewContext)
 
-            delegate?.conversationCreationController(
-                self,
-                didSelectName: values.name,
-                participants: values.participants,
+            let users = values.participants
+                .union([selfUser])
+                .materialize(in: userSession.viewContext)
+
+            service.createGroupConversation(
+                name: values.name,
+                users: Set(users),
                 allowGuests: values.allowGuests,
                 allowServices: values.allowServices,
                 enableReceipts: values.enableReceipts,
-                encryptionProtocol: values.encryptionProtocol
-            )
+                messageProtocol: values.encryptionProtocol == .proteus ? .proteus : .mls
+            ) { [weak self] in
+                guard let self = self else { return }
+
+                switch $0 {
+                case .success(let conversation):
+                    showNonFederatingDomainsAlert(domains: [
+                        "foo.com",
+                        "bar.com",
+                        "goo.com"
+                    ])
+                    return
+//                    delegate?.conversationCreationController(
+//                        self,
+//                        didCreateConversation: conversation
+//                    )
+
+                case .failure(let error):
+                    WireLogger.conversation.error("failed to create conversation: \(String(describing: error))")
+                }
+            }
         }
     }
+
+    private func showNonFederatingDomainsAlert(domains: Set<String>) {
+        typealias Strings = L10n.Localizable.Conversation.Create.NonFederatingDomainsError
+
+        let alert = UIAlertController(
+            title: Strings.title,
+            message: Strings.message(domains.sorted().localizedString()),
```

Comment:
In my opinion, `localizedString()` is too generic name. This name doesn't fully explain its purpose.

Reply:
Did you have something in mind? Any of these? `localizedList`, `joinAndLocalize`, `localizedJoin`.

Reply:
both `joinAndLocalize` or `localizedJoin` sounds good

Reply:
done